tests/std/tests/P0753R2_manipulators_for_cpp_synchronized_buffered_ostream/test.cpp

```diff
@@ -89,6 +92,76 @@ void test_osyncstream_manipulators(string_buffer<typename Ty::char_type, ThrowOn
     }
 }
 
+template <class Ty>
+class throwing_string_buffer : public basic_streambuf<Ty, char_traits<Ty>> {
+public:
+    throwing_string_buffer()  = default;
+    ~throwing_string_buffer() = default;
+
+    streamsize xsputn(const Ty*, streamsize) override {
+        throw std::ostream::failure("test exception");
+    }
+};
+
+void test_lwg_3571() {
+    // LWG-3571: "flush_emit should set badbit if the emit call fails"
+    {
+        osyncstream stream(nullptr);
+        stream << flush_emit;
+        assert(stream.rdstate() == ios::badbit);
+    }
+    {
+        stringstream inner;
+        osyncstream stream(inner);
+        stream << "Hello World";
+        stream.setstate(ios_base::failbit);
+        assert((stream.rdstate() & ios::badbit) == 0);

```

Comment:
As in #2523, I believe it would be clearer to test against `goodbit` instead of `0` here. Occurs below.